src/config/source/command_line_arguments.cpp

```diff
@@ -0,0 +1,158 @@
+   if (result == std::nullopt) {
+      throw silo::config::ConfigException(
+         fmt::format("the provided option '{}' is not a valid command line option", option)
+      );
+   }
+   return result.value();
+}
+
+namespace {
+
+// Split an individual command line argument in option and value
+// parts, if a '=' was given.
+std::tuple<std::string, std::optional<std::string>> splitOption(const std::string& arg) {
+   std::size_t pos = arg.find('=');
+   if (pos != std::string::npos) {
+      return {arg.substr(0, pos), arg.substr(pos + 1)};
+   }
+   return {arg, std::nullopt};
+}
+
+// Parse an option's value, taking the next argument if needed.
+std::tuple<ConfigValue, std::span<const std::string>> parseValueFromArg(
+   const ConfigAttributeSpecification& attribute_spec,
+   const std::string& arg,
+   const std::optional<std::string>& opt_value_string,
+   std::span<const std::string> remaining_args
+) {
+   if (attribute_spec.type == ConfigValueType::BOOL) {
+      if (opt_value_string.has_value()) {
+         throw silo::config::ConfigException("'=' not acceptable for boolean option: " + arg);
+      }
+      return {ConfigValue::fromBool(true), remaining_args};
+   }
+   std::string value_string;
+   if (opt_value_string.has_value()) {
+      value_string = opt_value_string.value();
+   } else {
+      if (remaining_args.empty()) {
+         throw silo::config::ConfigException("missing argument after option " + arg);
+      }
+      value_string = remaining_args[0];
+      remaining_args = remaining_args.subspan(1);
+   }
+   return {attribute_spec.parseValueFromString(value_string), remaining_args};
+}
+
+}  // namespace
+
+VerifiedConfigAttributes CommandLineArguments::verify(
+   const ConfigSpecification& config_specification
+) const {
+   // Now, given config_specification (and thus which options are
+   // boolean and which take arguments), we can parse the command
+   // line.
+
+   // E.g. "--api-foo" => "1234" or "--api-foo=1234"
+   std::unordered_map<ConfigKeyPath, ConfigValue> config_value_by_option;
+   std::vector<std::string> positional_args;
+   std::vector<std::string> invalid_config_keys;
+   std::span<const std::string> remaining_args{args.data(), args.size()};
+   while (!remaining_args.empty()) {
+      const std::string& arg = remaining_args[0];
+      remaining_args = remaining_args.subspan(1);
+      if (arg.starts_with('-')) {
+         if (arg == "--") {
+            std::ranges::copy(remaining_args, std::back_inserter(positional_args));
+            break;
+         }
+         if (arg == "-h" || arg == "--help") {
+            return {{}, {}, true};
+         }
+         const auto [option, opt_value_string] = splitOption(arg);
+         const auto ambiguous_key = stringToConfigKeyPath(option);
+         if (auto opt = config_specification.getAttributeSpecificationFromAmbiguousKey(ambiguous_key)) {
+            ConfigAttributeSpecification attribute_spec = opt.value();
```

Comment:
I find such situations ambiguous about what the best approach is. References are dangerous re life times, so it's generally good to avoid them. In Rust you'd write this as

    if let Some(attribute_spec) = config_specification.get_...(..) {
        ...
    }

and then if the get_ method returns a reference it's a reference (and not dangerous since borrow-checked), and if not, then it's moved to `attribute_spec` and lives there. Whereas here the `opt` variable is consciously written with a generic and short name because it is "just" a necessary temporary to achieve the pattern matching equivalent. Now should the value continue to live in that temporary variable? That contradicts that the variable is "temporary". Yes it lives long enough here, but besides the lifetime danger, would it also fail when moving the value somewhere else, as in `foo = std::move(attribute_spec)`? I guess it might actually work (I'm still fuzzy about these corners). At least if that doesn't work then 

    ConfigAttributeSpecification attribute_spec = std::move(opt.value());

seems to be the proper approach to me (and I've now changed it to that). Although `ConfigAttributeSpecification` doesn't currently have a move constructor; but then it won't matter, we're copying stuff left and right in the config library anyway.